### PR TITLE
cleanup unused scripts and Makefile

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: yeya24 <yb532204897@gmail.com>

### Ⅰ. Describe what this PR did
Clean up the unused scripts and Makefile.
1. make check is not used and we already have golangci-lint check
2. vendor is not used as well

### Ⅱ. Does this pull request fix one issue?

NONE

### CODE REVIEW

**Comment by Starnop** [comment]
@yeya24  Does golangci-lint also checks the boilerplate?

File: Makefile [hunk 1]
```diff
@@ -114,13 +114,6 @@ integration-test:  ## Run integration test
 	@go test ./test
 .PHONY: integration-test
 
-check: boilerplate-check  ## Check code formats
```
Comment (allencloud) [inline Makefile]: Actually this is used in https://github.com/dragonflyoss/Dragonfly/blob/9017881ccebde40b1ea47b875b71c4122b13333e/.circleci/config.yml#L59.

I do not think we could delete these.

File: hack/check-docker.sh [hunk 2]
```diff
@@ -1,14 +0,0 @@
-#!/bin/bash
```
Comment (allencloud) [inline hack/check-docker.sh]: I agree with this. We do not need to check whether docker is installed. How about only deleting :
```
@echo "Begin to check dockerd whether is startup"	
./hack/check-docker.sh
```
in the file above.

**Comment by allencloud** [comment]
> Does golangci-lint also checks the boilerplate?

I do not think so. @Starnop

File: Makefile (hunk 1 above)
Comment (yeya24) [reply Makefile]: I just removed the `check`, so we cannot use `make check` anymore.
But we can still use `make boilerplate-check` in ci. This does not conflict with my change.

File: hack/check-docker.sh (hunk 2 above)
Comment (yeya24) [reply hack/check-docker.sh]: check.sh is unnecessary as well. golangci-lint is the same of it.

**Comment by allencloud** [comment]
Oh, I agree with you. @yeya24

File: Makefile (hunk 1 above)
Comment (allencloud) [reply Makefile]: I agree. My misunderstanding.

**Comment by allencloud** [comment]
LGTM

**Pull request merged**